pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/creator/IndexCreationContext.java

```diff
@@ -0,0 +1,481 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.segment.spi.creator;
+
+import java.io.File;
+import java.util.Map;
+import java.util.Objects;
+import javax.annotation.Nullable;
+import org.apache.pinot.segment.spi.ColumnMetadata;
+import org.apache.pinot.segment.spi.compression.ChunkCompressionType;
+import org.apache.pinot.segment.spi.index.creator.H3IndexConfig;
+import org.apache.pinot.spi.config.table.BloomFilterConfig;
+import org.apache.pinot.spi.config.table.FSTType;
+import org.apache.pinot.spi.data.FieldSpec;
+
+
+/**
+ * Provides parameters for constructing indexes via {@see IndexCreatorProvider}.
+ * The responsibility for ensuring that the correct parameters for a particular
+ * index type lies with the caller.
+ */
+public interface IndexCreationContext {
+
+  FieldSpec getFieldSpec();
+
+  File getIndexDir();
+
+  boolean isOnHeap();
+
+  int getLengthOfLongestEntry();
+
+  int getMaxNumberOfMultiValueElements();
+
+  int getMaxRowLengthInBytes();
+
+  boolean isSorted();
+
+  int getCardinality();
+
+  int getTotalNumberOfEntries();
+
+  int getTotalDocs();
+
+  boolean hasDictionary();
+
+  BloomFilter forBloomFilter(BloomFilterConfig bloomFilterConfig);
+
+  Forward forForwardIndex(ChunkCompressionType chunkCompressionType,
+      @Nullable Map<String, Map<String, String>> columnProperties);
+
+  Geospatial forGeospatialIndex(H3IndexConfig h3IndexConfig);
+
+  Inverted forInvertedIndex();
+
+  Json forJsonIndex();
+
+  Range forRangeIndex(ColumnMetadata columnMetadata, int rangeIndexVersion);
+
+  Text forTextIndex(FSTType fstType, String[] sortedUniqueElementsArray);
```

Comment:
This is confusing. Will this support all 3 flavors of text index ?
- Lucene text
-Lucene FST
- Native FST

Reply:
I'll add another method to make the distinction between text and FST

Reply:
FWIW, the reason for this confusion is I just don't want to override text index construction and am only interested in forward, inverted, and JSON indexes, but the scope of this PR expanded to cover text. I don't fully understand the text indexing features and they're only included at all for completeness.

Reply:
Done.